src/sqlacodegen/cli.py

```diff
@@ -29,6 +30,26 @@
     from importlib.metadata import entry_points, version
 
 
+def _parse_engine_arg(arg_str: str) -> tuple[str, Any]:
+    if "=" not in arg_str:
+        raise argparse.ArgumentTypeError("engine-arg must be in key=value format")
+
+    key, value = arg_str.split("=", 1)
+    try:
+        value = ast.literal_eval(value)
+    except Exception:
+        pass  # Leave as string if literal_eval fails
+    return key, value
```

Comment:
Add a blank line before continuing after a control block ends.
```suggestion

    return key, value
```